### PR TITLE
feat: add checkbox to select all models/metrics/slices for a chart

### PR DESCRIPTION
**Pull request by Sparkier**
# Description

fix ZEN-204

### CODE REVIEW

**Comment by cabreraalex** [comment]
We need to get rid of the @lru_cache now since it won't update otherwise

File: backend/zeno_backend/processing/chart.py [hunk 1]
```diff
@@ -17,11 +17,74 @@
...
+    """Get the slices of the chart from a list of slice ids.
+
+    Args:
+        chart_slices (list[int]): slice ids of the chart.
+        project (str): project the user is currently working with.
+
+    Returns:
+        list[Slice]: list of slices of the chart.
+    """
+    if -2 in chart_slices:  # -2 signals that all slices should be selected
```
Comment (cabreraalex) [inline backend/zeno_backend/processing/chart.py:35]: Should this include `all_instances`? I can go either way.

File: backend/zeno_backend/processing/chart.py [hunk 2]
```diff
@@ -17,11 +17,74 @@
...
+    """Get the models of the chart from a list of models.
+
+    Args:
+        chart_models (list[str]): the list of models of the chart.
+        project (str): project the user is currently working with.
+
+    Returns:
+        list[str]: list of models of the chart.
+    """
+    if "all_zeno_project_models" in chart_models:  # all models should be selected
```
Comment (cabreraalex) [inline backend/zeno_backend/processing/chart.py:83]: what if someone names a model this ahaha

File: frontend/src/lib/components/chart/chart-page/encoding/chart-encoding/encoding-components/ModelsEncodingMultiChoice.svelte [hunk 3]
```diff
@@ -27,12 +28,26 @@
...
-	/>
+	<div class="flex flex-col">
+		{#if value[0] != 'all_zeno_project_models'}
+			<Svelecte
+				style="width: 280px; flex:none;"
+				bind:value
+				{options}
+				{dndzone}
+				multiple
+				placeholder="Select Slices..."
```
Comment (cabreraalex) [inline frontend/src/lib/components/chart/chart-page/encoding/chart-encoding/encoding-components/ModelsEncodingMultiChoice.svelte:39]: "Systems..."

File: frontend/src/lib/components/chart/chart-page/encoding/chart-encoding/encoding-components/ModelsEncodingMultiChoice.svelte [hunk 4]
```diff
@@ -27,12 +28,26 @@
...
+			/>
+		{/if}
+		<div class="flex items-center ml-auto">
+			<span>All Models</span>
```
Comment (cabreraalex) [inline frontend/src/lib/components/chart/chart-page/encoding/chart-encoding/encoding-components/ModelsEncodingMultiChoice.svelte:43]: "All Systems"

**Review by cabreraalex — CHANGES_REQUESTED** [review]

File: backend/zeno_backend/processing/chart.py (hunk 1 above)
Comment (Sparkier) [reply backend/zeno_backend/processing/chart.py:35]: Don't know either.... Often destroys the scales that's why I did not include it.

File: backend/zeno_backend/processing/chart.py (hunk 2 above)
Comment (Sparkier) [reply backend/zeno_backend/processing/chart.py:83]: Yep that's a problem. The alternative would be adding another field but don't really like that either. Don't know what to do....

File: backend/zeno_backend/processing/chart.py (hunk 1 above)
Comment (cabreraalex) [reply backend/zeno_backend/processing/chart.py:35]: Yeah I'm alright not including it.

File: backend/zeno_backend/processing/chart.py (hunk 2 above)
Comment (cabreraalex) [reply backend/zeno_backend/processing/chart.py:83]: does an empty string work?

**Comment by cabreraalex** [comment]
Can you also remove the @lru_cache

**Review by cabreraalex — APPROVED** [review]

File: backend/zeno_backend/processing/chart.py (hunk 2 above)
Comment (cabreraalex) [reply backend/zeno_backend/processing/chart.py:83]: if not it's fine super edge case

Comment (Sparkier) [reply backend/zeno_backend/processing/chart.py]: Very good idea!

**Pull request merged**